### PR TITLE
Predictable artifact order

### PR DESCRIPTION
**Pull request by jvierling**
On my system the integration test fails (at commit ea21cfe), because the order of the artifacts in the generated lockfile changes slightly. This pull request makes the order of the artifacts more predictable by ordering artifacts alphabetically. The pull request also includes some clean up commits.

### CODE REVIEW

File: src/main/java/com/fzakaria/mvn2nix/model/MavenNixInformation.java [hunk 1]
```diff
@@ -6,18 +6,15 @@
...
 
     private final Map<String, MavenArtifact> dependencies;
 
     public MavenNixInformation(Map<String, MavenArtifact> dependencies) {
-        this.dependencies = new HashMap<>(dependencies);
-    }
-
-    public Map<String, MavenArtifact> getDependencies() {
-        return new HashMap<>(dependencies);
+        this.dependencies = new TreeMap<>(dependencies);
```
Comment (fzakaria) [inline src/main/java/com/fzakaria/mvn2nix/model/MavenNixInformation.java:17]: Maybe add a comment explaining why it needs to be a TreeMap?

Did I have any test in this repo?
I can't remember :)

**Comment by jvierling** [comment]
@fzakaria Thanks for the review. I have added a comment and a unit test that shows that the artifacts are ordered alphabetically. The integration test Maven2nixIT also shows that the artifacts now appear in alphabetic order (see the modifications of the test's expected lockfile).

**Comment by jvierling** [comment]
FYI I fixed the Github-Actions on my fork (see https://github.com/jvierling/mvn2nix) by removing the `cachix/cachix-action`, updating the version of the `cachix/install-nix-action`, and removing the `install_url` from the flake jobs.

**Comment by fzakaria** [comment]
Do you want to include the updated github action as part of the PR?
or you can make it a separate PR.

**Comment by jvierling** [comment]
I'll include the update to the Github Actions as part of the PR. However, I am not sure if removing `cachix/cachix-action` is an option for you (I removed it on my fork because I do not have a cachix cache).

**Comment by fzakaria** [comment]
Thank you for the great PR.

**Pull request merged**